proto/viam/app/v1/app.proto

```diff
@@ -71,59 +146,111 @@ service AppService {
-  bool main_part = 8 [(tagger.v1.tags) = "bson:\"main_part\" json:\"main_part\""];
-  string fqdn = 9;
-  string local_fqdn = 11;
-  google.protobuf.Timestamp created_on = 13 [(tagger.v1.tags) = "bson:\"created_on\""];
+message Organization {
+  string id = 1;
+  string name = 2;
+  google.protobuf.Timestamp created_on = 3;
+}
 
-  // List of secrets allowed for authentication.
-  repeated SharedSecret secrets = 14 [(tagger.v1.tags) = "bson:\"secrets\""];
+message OrganizationInvite {
+  string organization_id = 2;
+  string email = 3;
+  google.protobuf.Timestamp created_on = 4;
 }
 
-message RobotPartHistoryEntry {
-  string part = 1 [(tagger.v1.tags) = "bson:\"part\" json:\"part\""];
-  string robot = 2 [(tagger.v1.tags) = "bson:\"robot\" json:\"robot\""];
-  google.protobuf.Timestamp when = 3 [(tagger.v1.tags) = "bson:\"when\" json:\"when\""];
-  RobotPart old = 4 [(tagger.v1.tags) = "bson:\"old\" json:\"old\""];
+message CreateOrganizationRequest {
+  string name = 1;
+}
+
+message CreateOrganizationResponse {
+  Organization organization = 1;
 }
 
 message ListOrganizationsRequest {}
 
-message Organization {
-  string id = 1;
+message ListOrganizationsResponse {
+  repeated Organization organizations = 1;
+}
+
+message GetOrganizationRequest {
+  string organization_id = 1;
+}
+
+message GetOrganizationResponse {
+  Organization organization = 1;
+}
+
+message UpdateOrganizationRequest {
+  string organization_id = 1;
   string name = 2;
-  google.protobuf.Timestamp created_on = 3;
 }
 
-message ListOrganizationsResponse {
-  repeated Organization organizations = 1;
+message UpdateOrganizationResponse {
+  Organization organization = 1;
+}
+
+message DeleteOrganizationRequest {
+  string organization_id = 1;
+}
+
+message DeleteOrganizationResponse {}
+
+message ListOrganizationMembersRequest {
+  string organization_id = 1;
+}
+
+message ListOrganizationMembersResponse {
+  string organization_id = 1;
+  repeated OrganizationMember members = 2;
+  repeated OrganizationInvite invites = 3;
+}
+
+message CreateOrganizationInviteRequest {
+  string organization_id = 1;
+  string email = 2;
+}
+
+message CreateOrganizationInviteResponse {
+  OrganizationInvite invite = 1;
+}
+
+message DeleteOrganizationMemberRequest {
+  string organization_id = 1;
+  string user_id = 2;
+}
+
+message DeleteOrganizationMemberResponse {}
+
+message DeleteOrganizationInviteRequest {
+  string organization_id = 1;
+  string email = 2;
+}
+
+message DeleteOrganizationInviteResponse {}
+
+// Location
+//
+
+message LocationOrganization {
+  string organization_id = 1;
+  bool primary = 2;
 }
 
 message Location {
   string id = 1;
   string name = 2;
-  google.protobuf.Timestamp created_on = 3;
+  string parent = 3;
+  LocationAuth auth = 4;
+  bool assigned = 5;
+  google.protobuf.Timestamp created_on = 6;
 }
 
 message CreateLocationRequest {
```

Comment:
If we added `parent` we could allow API users to chose a parent on creation instead of having to make an update call right afterwards.